src/iden3comm/handlers/payment.ts

```diff
@@ -47,15 +55,125 @@ export function createPaymentRequest(
   return request;
 }
 
+/**
+ * @beta
+ * createPaymentRailsV1 is a function to create protocol payment message
+ * @param {DID} sender - sender did
+ * @param {DID} receiver - receiver did
+ * @param {Signer} signer - receiver did
+ * @param {string} agent - agent URL
+ * @param opts - payment options
+ * @returns {Promise<PaymentRequestMessage>}
+ */
+export async function createPaymentRailsV1(
+  sender: DID,
+  receiver: DID,
+  agent: string,
+  signer: Signer,
+  opts: {
+    payments: [
+      {
+        credentials: {
+          type: string;
+          context: string;
+        }[];
+        description?: string;
+        chains: {
+          nonce: bigint;
+          amount: bigint;
+          currency: SupportedCurrencies;
+          chainId: string;
+          recipient: string;
+          verifyingContract: string;
+          expirationDate?: Date;
+        }[];
+      }
+    ];
+  }
+): Promise<PaymentRequestMessage> {
+  const payments: PaymentRequestInfo[] = [];
+  for (let i = 0; i < opts.payments.length; i++) {
+    const { credentials, description } = opts.payments[i];
+    const dataArr: Iden3PaymentRailsRequestV1[] = [];
+    for (let j = 0; j < opts.payments[i].chains.length; j++) {
+      const { nonce, amount, currency, chainId, recipient, verifyingContract, expirationDate } =
+        opts.payments[i].chains[j];
+
+      if (recipient !== (await signer.getAddress())) {
+        throw new Error('recipient is not the signer');
+      }
+      const typeUrl = 'https://schema.iden3.io/core/json/Iden3PaymentRailsRequestV1.json';
+      const typesFetchResult = await fetch(typeUrl);
+      const types = await typesFetchResult.json();
+      delete types.EIP712Domain;
+      const paymentData = {
+        recipient,
+        amount,
+        expirationDate: expirationDate?.getTime() ?? 0,
+        nonce,
+        metadata: '0x'
+      };
+
+      const domain = {
+        name: 'MCPayment',
```

Comment:
what is MCPayment?

Reply:
multichain payment, name that we passed in contract as domain name (part of the EIP712Domain)

Reply:
__EIP712_init("MCPayment", VERSION);
from contract - https://github.com/iden3/contracts/blob/master/contracts/payment/MCPayment.sol#L76